lib/hostip.c

```diff
@@ -494,6 +494,49 @@ enum resolve_t Curl_resolv(struct connectdata *conn,
   CURLcode result;
   enum resolve_t rc = CURLRESOLV_ERROR; /* default to failure */
 
+  /* Check for illegal Top-Level-Domain */
+  {
+    const char ILLEGAL_TLDS[2][10] = {
+      "onion",
+      "i2p"
```

Comment:
Honestly I don't know but it is the biggest alternative to Tor and probably suffers from the same problem

Reply:
Maybe, but then they should go the `.onion` route (no pun intended) and request the same treatment so that we as a community would act similarly on it. I don't think it is curl's job to run ahead of the pack and voluntarily ignore TLDs someone created somewhere. I want a stronger signal that this is the right thing. Before your PR here, I had never even heard of i2p.

Reply:
> Before your PR here, I had never even heard of i2p.

We learn every day something new ;)
But I agree on the point that it is not curls job to standardize something. I'll remove i2p but would like to continue using an array for illegal TLDs for the case we need more in future